mleap-core/src/main/scala/ml/combust/mleap/core/types/Casting.scala

```diff
@@ -184,13 +184,41 @@ object Casting {
                   val cc = c.asInstanceOf[(Any) => ByteString]
                   (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(cc).toArray)
               }
+            }
+          }
+        }.orElse {
+          Some {
+            Try {
+              from.base match {
+                case BasicType.Boolean =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Boolean]).toArray)
+                case BasicType.Byte =>
+                  (v: Any) => Tensor.denseVector(v.asInstanceOf[Seq[_]].map(_.asInstanceOf[Byte]).toArray)
```

Comment:
It's annoying to have to manually have specify these `asInstanceOf` calls when we don't actually want to cast the elements, but it is a consequence of making the function input `(v: Any)` and casting it to `Seq[_]`

If we don't manually re-cast each element, it will make a tensor of Java `Object` types, and our tests fail